### ISSUE TITLE
Package Kata Containers for Fedora s390x

### ISSUE THREAD

**Issue by Jakob-Naucke** [kata-containers/kata-containers#2632]
@etrunko we originally spoke way back in #1204, and since then, #2385 has also been released, so I think s390x should work. [It's not yet being built](https://koji.fedoraproject.org/koji/buildinfo?buildID=1829173), can you TAL?

**Comment by fidencio** [comment]
JFYI, I've removed `s390x` from the "do-not-build" list and fired a scratch-build: https://koji.fedoraproject.org/koji/taskinfo?taskID=75657307

That may give us some ideas on whether we'd be good to just ship it or whether we'd need to do any kind of changes in the Fedora's spec file.

**Comment by fidencio** [comment]
And we now can build the package for s390x, with the following change:
```
fidencio@quino ~/src/fedora/kata-containers $ git diff
diff --git a/kata-containers.spec b/kata-containers.spec
index 01d8b3a..bb34e54 100644
--- a/kata-containers.spec
+++ b/kata-containers.spec
@@ -158,8 +158,7 @@ Conflicts: kata-runtime
 Conflicts: kata-shim
 
 # The following architectures lack the required qemu support
-# s390 fail to build: https://github.com/kata-containers/kata-containers/issues/1204
-ExcludeArch: %{arm} %{ix86} s390 s390x
+ExcludeArch: %{arm} %{ix86} s390
 
 
 %description
```

Do we care about s390 as well, @Jakob-Naucke?  (Sorry for the ignorance on my side).

Last but not least, would you be willing to send a patch to https://src.fedoraproject.org/rpms/kata-containers with the change described above + the adjusting NVR and whatnot?  I could guide you through the process if that's of your interest.

**Comment by etrunko** [comment]
> @etrunko we originally spoke way back in #1204, and since then, #2385 has also been released, so I think s390x should work. [It's not yet being built](https://koji.fedoraproject.org/koji/buildinfo?buildID=1829173), can you TAL?

Sorry, I was not following up this closely, it can be enabled for sure. Can you do a quick test with the scratch build @fidencio provided?

**Comment by fidencio** [comment]
@Jakob-Naucke, mind that with the scratch-build provided you may hit https://github.com/kata-containers/kata-containers/issues/2539.  This is going to be solved with the 2.2.1 release coming out next Friday.

**Comment by Jakob-Naucke** [comment]
> And we now can build the package for s390x

@fidencio just tested, it works! :tada:

> Do we care about s390 as well, @Jakob-Naucke? (Sorry for the ignorance on my side).

s390 is 32bit (31bit addresses). The last piece of (stricly) s390 hardware was released in 1998 AFAICT and Fedora only packages s390x, so unless I'm misunderstanding something about Fedora's naming conventions (e.g. kernel still calls the entire tree `s390`), no :grin:

> Last but not least, would you be willing to send a patch to https://src.fedoraproject.org/rpms/kata-containers with the change described above + the adjusting NVR and whatnot? I could guide you through the process if that's of your interest.

I'll TAL, but I might need some help.

**Comment by Jakob-Naucke** [comment]
for reference: https://src.fedoraproject.org/rpms/kata-containers/pull-request/3

**Comment by Jakob-Naucke** [comment]
was merged for 2.2.0-3 -- thanks @etrunko @fidencio!
